app/api/sensors/route.js

```diff
@@ -18,29 +17,30 @@ export async function GET() {
     },
   })
 
+  // Check if sensors exist, if not, redirect to a 404 page
+  if (!sensors || sensors.length === 0) {
+    return NextResponse.redirect('/404') // Adjust the path to your actual 404 page
```

Comment:
As it an api maybe, we should return a code instead redirect : `NextResponse.json(sensors, { status: 404 })`. When can manage the respond code into the frontend